instance/main.tf

```diff
@@ -72,6 +85,6 @@ resource "aws_instance" "web_private" {
 
 
   tags = {
-    Name = "${var.env_code}-web-private"
+    Name = "${data.terraform_remote_state.networking.outputs.env_code}-web-private"
```

Comment:
Please don't use a remote state for this variable. Instead, declare it as a separate variable in level2.
The name can remain the same: level2